projects/observability/src/pages/explorer/explorer.component.ts

```diff
@@ -79,25 +86,34 @@ import {
   `
 })
 export class ExplorerComponent {
-  public static readonly API_TRACES: string = 'Endpoint Traces';
-  public static readonly SPANS: string = 'Spans';
+  public static readonly API_TRACES: 'Endpoint Traces' = 'Endpoint Traces';
+  public static readonly SPANS: 'Spans' = 'Spans';
+  private static readonly SCOPE_QUERY_PARAM: string = 'scope';
 
   private readonly explorerDashboardBuilder: ExplorerDashboardBuilder;
   public readonly resultsDashboard$: Observable<ExplorerGeneratedDashboard>;
   public readonly vizDashboard$: Observable<ExplorerGeneratedDashboard>;
+  public readonly contextLabel$: Observable<ContextLabel>;
 
-  public context: ExplorerGeneratedDashboardContext = ObservabilityTraceType.Api;
+  public context?: ExplorerGeneratedDashboardContext;
   public filters: Filter[] = [];
 
   public visualizationExpanded: boolean = true;
   public resultsExpanded: boolean = true;
 
   public constructor(
-    @Inject(EXPLORER_DASHBOARD_BUILDER_FACTORY) explorerDasboardBuilderFactory: ExplorerDashboardBuilderFactory
+    private readonly navigationService: NavigationService,
+    @Inject(EXPLORER_DASHBOARD_BUILDER_FACTORY) explorerDasboardBuilderFactory: ExplorerDashboardBuilderFactory,
+    activatedRoute: ActivatedRoute
   ) {
     this.explorerDashboardBuilder = explorerDasboardBuilderFactory.build();
     this.resultsDashboard$ = this.explorerDashboardBuilder.resultsDashboard$;
     this.vizDashboard$ = this.explorerDashboardBuilder.visualizationDashboard$;
+    this.contextLabel$ = activatedRoute.queryParamMap.pipe(
+      map(paramMap => paramMap.get(ExplorerComponent.SCOPE_QUERY_PARAM)),
+      map(queryParam => this.queryParamToContextLabel(queryParam)),
+      tap(label => this.onContextUpdated(label))
```

Comment:
Keeping this here wouldn't update URL again? Since the change is coming from URL, why call this function here?

Reply:
Technically it would, but it's idempotent. The first time through, we do want it to update the URL to apply the default, it's just on any later url changes where it becomes unnecessary. I fixed it in one of the iterations, but felt like the extra complexity to avoid the dupe calls wasn't worth it.